capsules/src/lsm303agr.rs

```diff
@@ -496,8 +523,25 @@ impl Driver for Lsm303agrI2C<'_> {
         command_num: usize,
         data1: usize,
         data2: usize,
-        _appid: ProcessId,
+        process_id: ProcessId,
     ) -> CommandReturn {
+        if command_num == 0 {
+            // Handle this first as it should be returned
+            // unconditionally
+            return CommandReturn::success();
+        }
+
+        let match_or_empty_or_nonexistant = self.owning_process.map_or(true, |current_process| {
+            self.apps
+                .enter(*current_process, |_| current_process == &process_id)
+                .unwrap_or(true)
+        });
+        if match_or_empty_or_nonexistant {
+            self.owning_process.set(process_id);
+        } else {
+            return CommandReturn::failure(ErrorCode::RESERVE);
+        }
+
         match command_num {
             0 => CommandReturn::success(),
```

Comment:
This line of the match could be removed since it is handled above, but that is just a nit -- pretty sure the compiler will elide that code either way.

Reply:
Fixed.